src/wallet/rpcwallet.cpp

```diff
@@ -1198,15 +1198,13 @@ UniValue proposeanswer(const UniValue& params, bool fHelp)
     if(!consultation.CanHaveNewAnswers())
         throw JSONRPCError(RPC_TYPE_ERROR, "The consultation does not admit new answers.");
 
-    int64_t nValue = params[1].get_int64();
+    std::string sAnswer = params[1].get_str();
```

Comment:
I propose using this as a fix:

```c++
    std::string sAnswer = "";
    if (consultation.IsAboutConsensusParameter())
    {
        int64_t nValue = params[1].get_int64();

        if (consultation.nMin == Consensus::CONSENSUS_PARAM_PROPOSAL_MAX_VOTING_CYCLES || consultation.nMin == Consensus::CONSENSUS_PARAM_PAYMENT_REQUEST_MAX_VOTING_CYCLES)
        {
            nValue--;
        }

        sAnswer = std::to_string(nValue);
    }
    else
    {
        sAnswer = params[1].get_str();
    }
```

nMin could be equal to Consensus::CONSENSUS_PARAM_PROPOSAL_MAX_VOTING_CYCLES or Consensus::CONSENSUS_PARAM_PAYMENT_REQUEST_MAX_VOTING_CYCLES for non consensus consultations and trigger this erroneously.

Reply:
I've pushed the change